test/system/helpers.network.bash

```diff
@@ -241,6 +241,13 @@ function port_is_bound() {
         local proto="tcp"
     fi
 
+    # /proc/net/tcp is insufficient: it does not show ports in namespaces.
+    # ss does, so check it first.
+    run ss -${proto:0:1}nlH sport = $port
+    if [[ -n "$output" ]]; then
+        return
+    fi
```

Comment:
Why do you think ss shows ports in a different network namespace? I cannot see anything indicating that it does.
There is a `--net=NSNAME` that would make it switch the namespace.

More importantly checking the host port in the namespace would make no sense, you would need to check the target port (i.e. 80 for the webserver)

If you actually care about the netns something like this will be needed: https://github.com/containers/netavark/commit/91f3bfbfba6d69f791cec237d7c87ea8cb9a8d1e

Reply:
I have no idea. I'm just reporting the facts. Try the example in my commit message.

Reply:
Well I tried ss and it does not report the port and it makes zero sense for it to do it, it would completely break the point of network namespaces.
terminal 1:
```
$ sudo unshare -n sh -c 'ip link set lo up && nc -vl 0.0.0.0 8080'
Ncat: Version 7.93 ( https://nmap.org/ncat )
Ncat: Listening on 0.0.0.0:8080
```
terminal 2:
```
sudo ss -nltp sport = 8080
State                      Recv-Q                      Send-Q                                           Local Address:Port                                           Peer Address:Port                     Process                     
$ sudo nsenter -n -t $(pgrep nc) ss -nltp 
State                      Recv-Q                      Send-Q                                           Local Address:Port                                           Peer Address:Port                     Process                     
LISTEN                     0                           10                                                     0.0.0.0:8080                                                0.0.0.0:*                         users:(("nc",pid=17056,fd=3))
```

Only if you are part of the netns you can see the bound ports.

Reply:
It's a terminology thing, then. I assumed the problem was "network namespaces", maybe it's something else. I've re-pushed with a vaguer commit message and comment. Regardless, though, the fact remains that `ss` finds things that `/proc/net/tcp` does not.

Reply:
Correct that might be the case
AFAIK /proc/net/tcp{,6} is deprecated. I think the thing is that it does not work with proper dualstack bind.
You can compare `netstat` vs `ss`, the former reads the /proc/net... while the later uses the new netlink kernel API for it (https://man7.org/linux/man-pages/man7/sock_diag.7.html)

That leaves the question why keep the /proc code around and not just only depend on `ss`, I mean with this code it should be practically impossible for ss to not find the port while /proc/net/tcp does.

Reply:
Excellent - thank you for the context, and my sheepish apologies for making a stupid assumption.

>why keep the /proc code around

From my perspective, because the complicated `/proc/net` code is not mine and I don't have the skilz to figure out if it might catch something that my quick-hack `ss` does not. @sbrivio-rh that's your code, WDYT?